### PR TITLE
Fix make cross build failure

### LINKED ISSUE #50675: The cross-build is broken
```
I0815 09:30:09.341] +++ [0815 09:28:58] windows/amd64: go build started
I0815 09:30:09.341] # k8s.io/kubernetes/pkg/kubelet/dockershim
I0815 09:30:09.341] pkg/kubelet/dockershim/docker_container.go:153: not enough arguments in call to ds.getSecurityOpts
I0815 09:30:09.341] 	have (string, rune)
I0815 09:30:09.342] 	want (string, *"k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime".PodSandboxConfig, rune)
I0815 09:30:09.342] pkg/kubelet/dockershim/docker_sandbox.go:591: not enough arguments in call to ds.getSecurityOpts
I0815 09:30:09.342] 	have (string, rune)
I0815 09:30:09.342] 	want (string, *"k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime".PodSandboxConfig, rune)
I0815 09:30:09.342] !!! [0815 09:30:09] Call tree:
I0815 09:30:09.343] !!! [0815 09:30:09]  1: /go/src/k8s.io/kubernetes/hack/lib/golang.sh:697 kube::golang::build_binaries_for_platform(...)
I0815 09:30:09.343] !!! [0815 09:30:09]  2: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
I0815 09:30:09.348] Makefile:92: recipe for target 'all' failed
I0815 09:30:09.351] Makefile:419: recipe for target 'cross' failed
W0815 09:30:09.452] !!! [0815 09:30:09] Call tree:
W0815 09:30:09.452] !!! [0815 09:30:09]  1: hack/make-rules/build.sh:27 kube::golang::build_binaries(...)
W0815 09:30:09.452] make[1]: *** [all] Error 1
W0815 09:30:09.453] make: *** [cross] Error 1
W0815 09:30:09.743] !!! [0815 09:30:09] Call tree:
W0815 09:30:09.745] !!! [0815 09:30:09]  1: build/../build/common.sh:521 kube::build::run_build_command_ex(...)
W0815 09:30:09.747] !!! [0815 09:30:09]  2: build/release.sh:36 kube::build::run_build_command(...)
W0815 09:30:09.760] make: *** [release] Error 1
W0815 09:30:09.761] Traceback (most recent call last):
W0815 09:30:09.761]   File "/var/lib/jenkins/workspace/pull-kubernetes-cross/./test-infra/jenkins/../scenarios/execute.py", line 50, in <module>
W0815 09:30:09.784]     main(ARGS.env, ARGS.cmd + ARGS.args)
W0815 09:30:09.784]   File "/var/lib/jenkins/workspace/pull-kubernetes-cross/./test-infra/jenkins/../scenarios/execute.py", line 41, in main
W0815 09:30:09.785]     check(*cmd)
W0815 09:30:09.785]   File "/var/lib/jenkins/workspace/pull-kubernetes-cross/./test-infra/jenkins/../scenarios/execute.py", line 30, in check
W0815 09:30:09.785]     subprocess.check_call(cmd)
W0815 09:30:09.785]   File "/usr/lib/python2.7/subprocess.py", line 186, in check_call
W0815 09:30:09.802]     raise CalledProcessError(retcode, cmd)
W0815 09:30:09.802] subprocess.CalledProcessError: Command '('make', 'release')' returned non-zero exit status 2
I0815 09:30:09.809] Makefile:382: recipe for target 'release' failed
E0815 09:30:09.809] Command failed
```

seems to belong to sig node @kubernetes/sig-node-bugs PTAL and fix ASAP

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

had to fix the method getSecurityOpts in helpers_windows.go to
match the implementation in helpers_linux.go from commit:
bf01fa2f0010ae02d1ef3369ea268a2b4241619d

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

Fixes #50675

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by luxas — COMMENTED** [review]
Don't have enough context on this to fully LGTM, but seems reasonable to be to just no-op on windows that doesn't support this feature...

File: pkg/kubelet/dockershim/helpers_windows.go [hunk 1]
```diff
@@ -26,27 +26,16 @@ import (
...
-		hasSeccompSetting = true
-	}
-
-	if hasSeccompSetting {
-		glog.Warningf("seccomp annotations found, but it is not supported on windows")
-	}
-
+func (ds *dockerService) getSecurityOpts(seccompProfile string, separator rune) ([]string, error) {
+	// return nil so docker will load the default seccomp profile
+	glog.Warningf("seccomp annotations are not supported on windows")
```
Comment (yujuhong) [inline pkg/kubelet/dockershim/helpers_windows.go]: Warning should only be logged if there are annotations. Removing the condition will lead to false warnings.

**Review by yujuhong — CHANGES_REQUESTED** [review]

File: pkg/kubelet/dockershim/helpers_windows.go (hunk 1 above)
Comment (dims) [inline pkg/kubelet/dockershim/helpers_windows.go]: @yujuhong : did you see this? https://github.com/kubernetes/kubernetes/blob/b32639f9e7d47d0f69bb41b5535803ae4c9d835a/pkg/kubelet/dockershim/helpers_unsupported.go#L33

With the recent changes, the first parameter to that method is the `seccompProfile`, not `containerName` Also, please look at how the calls to getSecurityOpts are made from docker_container.go and docker_sandbox.go

Thanks,
Dims

Comment (Random-Liu) [inline pkg/kubelet/dockershim/helpers_windows.go]: Maybe only log the warning when `seccompProfile` is not empty.

Comment (dims) [inline pkg/kubelet/dockershim/helpers_windows.go]: Done @Random-Liu

File: pkg/kubelet/dockershim/helpers_windows.go [hunk 2]
```diff
@@ -26,27 +26,18 @@ import (
...
+func (ds *dockerService) getSecurityOpts(seccompProfile string, separator rune) ([]string, error) {
+	if seccompProfile != "" {
+		glog.Warningf("seccomp annotations are not supported on windows")
 	}
-
-	if hasSeccompSetting {
-		glog.Warningf("seccomp annotations found, but it is not supported on windows")
-	}
-
+	// return nil so docker will load the default seccomp profile
```
Comment (yujuhong) [inline pkg/kubelet/dockershim/helpers_windows.go]: Is this comment true for windows containers?

File: pkg/kubelet/dockershim/helpers_windows.go (hunk 1 above)
Comment (yujuhong) [inline pkg/kubelet/dockershim/helpers_windows.go]: @Random-Liu's suggestion LGTM. I was going to suggest the same thing :-)

File: pkg/kubelet/dockershim/helpers_windows.go (hunk 2 above)
Comment (dims) [inline pkg/kubelet/dockershim/helpers_windows.go]: @yujuhong comment is true to the best of my knowledge. you can actually see it in helpers_linux.go as well

Comment (yujuhong) [inline pkg/kubelet/dockershim/helpers_windows.go]: Uh..how is seccomp supported on windows?

**Review by yujuhong — APPROVED** [review]

**Comment by dims** [comment]
Thanks @yujuhong

**Comment by feiskyer** [comment]
@dims Thanks for the fix.

**Pull request merged**